core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/SampleQCImpl.java

```diff
@@ -135,4 +136,22 @@ public int hashCode() {
       return hashcode;
     }
   }
+
+  @Override
+  public int compareTo(QC t) {
+    SampleQC s = (SampleQC) t;
+    if (getId() != 0L && t.getId() != 0L) {
+      if (getId() < s.getId()) return -1;
+      if (getId() > s.getId()) return 1;
+    } else if (getQcType() != null && s.getQcType() != null && getResults() != null && s.getResults() != null && getQcDate() != null
+        && s.getQcDate() != null) {
+      int type = getQcType().compareTo(s.getQcType());
+      if (type != 0) return type;
+      int results = getResults().compareTo(s.getResults());
+      if (results != 0) return results;
+      int creator = getQcDate().compareTo(s.getQcDate());
+      if (creator != 0) return creator;
+    }
```

Comment:
These compareTo(QC) could be abstracted a bit: If it's okay to compare date before data, just make the impl classes do `int compare = super.compareTo(t); if (compare == 0) {// impl-specific compare} else {return 0}`

If date really should be compared after data, make AbstractQc do the generic stuff, then call an abstract "compareQcData" method, which is overridden in all of the impl classes to only compare implementation-specific things. If that's still 0, do the date comparison.

May want to move null checks around too (it doesn't matter if qcDate is null if it ends up basing the response on qcType)

Just thoughts, no change necessarily needed. As is, it's better than before